### ISSUE TITLE
Per gamemode type bypass permissions

### ISSUE THREAD

**Issue by dc6iq** [Multiverse/Multiverse-Core#2547]
While being in spectator gamemode, you can teleport to a spectated player using numeric keys. If the spectated player is in
another world, your gamemode is being changed from spectator to the word's mode. If the spectated player is flying, you will be in
air as well and most likely fall down and die because your gamemode was being changed.

### Feature request

** If a permission mv.bypass.onspectator.* is set and you are in spectator mode, no gamemode change will happen.

**Feature description**
a new permission set mv.bypass.onspectator.world / mv.bypass.onspectator.* should be added. If it is set _and_ your current gamemode is spectator, this gamemode will be kept while being teleported to another world.

**How the feature is useful**
You won't die anymore and you will not be spotted if you traverse a worlds in spectator mode.

** Fix **
Just pull the fix from https://github.com/dc6iq/Multiverse-Core.git, thank you

**Comment by dc6iq** [comment]
Pull request added...

**Comment by benwoo1110** [comment]
There is already `mv.bypass.gamemode.WORLDNAME`. See: https://github.com/Multiverse/Multiverse-Core/wiki/Big-List-O'-Permissions#access-permissions

Only thing is if you want only spectator to bypass gamemode on world change but not other gamemodes. For this, more changes will be needed.

**Comment by dc6iq** [comment]
This does too much.
if i am in survival mode, then i am just a normal player, so a gamemode change should happen.

**Comment by benwoo1110** [comment]
Essentially what you are looking for is per gamemode type bypass permissions, which is a good suggestion.

My concerns are 
1. If it were to be added, I think we should have permission node for all gamemodes, not just spectator as well.
2. `mv.bypass.gamemode.<gamemode>.<worldname>` may be a better perm node then
3. And in all honestly, we need a full permission revamp general lol

Let see other devs opinion first, @nicegamer7 @dumptruckman

**Comment by willkroboth** [comment]
I feel like you should be able to use contexts from a permissions plugin like LuckPerms to do this. If you just turn on mv.bypass.gamemode.WORLDNAME when your gamemode is spectator it should do what you want.

**Comment by nicegamer7** [comment]
I agree with your third point ben, and what willkroboth suggests should also work.

**Comment by zax71** [comment]
I think MV shouldn’t handle this as LuckPerms can already do it with what I’d think of as a much more intuitive method.

**Comment by benwoo1110** [comment]
Closing this -I think LuckPerms context is a good enough solution
